core/session.ts

```diff
@@ -515,6 +518,12 @@ export class Session {
       action.dependencies = (action.dependencyTargets || []).map(dependencyTarget =>
         utils.targetToName(dependencyTarget)
       );
+
+      if ("parent" in action) {
```

Comment:
why not just change `IActionProto` to include an optional `parent` field, revert the changes to the method declaration, and then get rid of this?

Reply:
Sure that seems neater